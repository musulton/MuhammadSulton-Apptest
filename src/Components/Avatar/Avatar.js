import Icon from 'react-native-vector-icons/FontAwesome';
import * as React from 'react';
import {Image, View} from 'react-native';

import {validURL} from '../../Utils/Url';
import Colors from '../../Themes/Colors';

import styles from './Avatar.styles';

const Avatar = ({uri, size, fromGallery}) => {
  const shouldDisplayImage = fromGallery ? uri : validURL(uri);
  if (shouldDisplayImage) {
    return <Image source={{uri}} style={styles.avatarBackground(size)} />;
  }

  return (
    <View style={styles.avatarBackground(size)}>
      <Icon name="user" size={40} color={Colors.icon} />
    </View>
  );
};

export default Avatar;
