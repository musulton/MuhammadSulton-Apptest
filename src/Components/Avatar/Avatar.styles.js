import {StyleSheet} from 'react-native';

import Colors from '../../Themes/Colors';

export default StyleSheet.create({
  avatarBackground: size => ({
    width: size,
    height: size,
    borderRadius: 50,
    backgroundColor: Colors.avatar,
    justifyContent: 'center',
    alignItems: 'center',
  }),
});
