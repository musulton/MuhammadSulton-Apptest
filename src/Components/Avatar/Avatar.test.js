import * as React from 'react';
import {render, cleanup} from '@testing-library/react-native';

import Avatar from './Avatar';

describe('Avatar', () => {
  afterEach(cleanup);

  const props = {
    uri: 'N/A',
    size: 30,
    fromGallery: false,
  };

  it('should render avatar without image correctly', () => {
    const {toJSON} = render(<Avatar {...props} />);

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render avatar with image correctly', () => {
    const _props = {
      ...props,
      uri: 'https://static.wikia.nocookie.net/naruto/images/d/dd/Naruto_Uzumaki%21%21.png/revision/latest?cb=20161013233552',
    };
    const {toJSON} = render(<Avatar {..._props} />);

    expect(toJSON()).toMatchSnapshot();
  });
});
