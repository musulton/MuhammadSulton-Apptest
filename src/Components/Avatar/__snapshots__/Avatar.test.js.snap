// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render avatar with image correctly 1`] = `
<Image
  source={
    Object {
      "uri": "https://static.wikia.nocookie.net/naruto/images/d/dd/Naruto_Uzumaki%21%21.png/revision/latest?cb=20161013233552",
    }
  }
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9DE0AD",
      "borderRadius": 50,
      "height": 30,
      "justifyContent": "center",
      "width": 30,
    }
  }
/>
`;

exports[`Avatar should render avatar without image correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#9DE0AD",
      "borderRadius": 50,
      "height": 30,
      "justifyContent": "center",
      "width": 30,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 40,
        },
        undefined,
        Object {
          "fontFamily": "FontAwesome",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï€‡
  </Text>
</View>
`;
